test/unit/client_interface_sections/search_test.rb

```diff
@@ -21,4 +21,13 @@ def test_url_encoding_only_happens_once
     assert_equal 'search-test/Appointment?date=%3E2016-01-01&patient=test',
                  reply.request[:url]
   end
+
+  def test_url_encoded_form
+    stub_request(:post, /search-test/).to_return(body: '{"resourceType":"Bundle"}')
+    headers = client.fhir_headers(content_type: 'application/x-www-form-urlencoded')
+    payload = {test: "test-body"}
+    reply = client.send(:post, '/search-test', payload, headers)
```

Comment:
Is there a reason for invoking the private method `#post` with `#send` instead of using the `#search` method?  See the existing test in this file for an example of that.

It is currently under-documented and the API could use some updates, but if you set the flag parameter to true you can search with a POST and you won't need to manually set the Content-Type header.  I think it would be preferable to use (and improve) the abstractions (read, create, search, etc.) over the base http methods (get, post, put, etc.).

See the implementation here: https://github.com/fhir-crucible/fhir_client/blob/master/lib/fhir_client/sections/search.rb#L16

Reply:
It sounds like this existing param is what @rtjhie0809 is looking for.  Though to say this is a cryptic name is an understatement.  I wonder if we can add another term in there that is more descriptive (like `:form_post` or something), and deprecate the unhelpful `:flag` parameter?

Reply:
Oh you're absolutely right. That's the search post I'm looking for. Apologies for the lack of research on my part. Unfortunately what it appears to be missing is the ability to provide some search parameters in the body.

Reply:
Hear, hear.  There are a few other search methods that would need the update applied as well.  I do think a more comprehensive assessment of the API should be done, but this would be a good update in the meantime.

Bonus points if we update the docs so that others are aware of the search by POST functionality in the future :)

Reply:
Hah, yearh, I think the existing behavior is completely wrong.  It makes no sense to encode the parameters in the querystring when the method is a POST.  The whole point of this is to allow users to execute a search from filling in a form on the browser.  I feel like this is a bug and we should just fix it.  (though we definitely shouldn't make a breaking change for `:flag` until a new major version, which won't be until r5 comes out).  Just my 2c.

Reply:
Yeah, it appears like it could be possible to add the body parameter to `#search` by replacing the `nil` with something like `options[:body]`.